src/base/bittorrent/torrentinfo.h

```diff
@@ -31,19 +31,18 @@
 
 #include <libtorrent/torrent_info.hpp>
 
+#include <QByteArray>
 #include <QCoreApplication>
+#include <QDateTime>
 #include <QList>
 #include <QtGlobal>
+#include <QString>
+#include <QStringList>
 #include <QVector>
+#include <QUrl>
 
 #include "base/indexrange.h"
 
-class QByteArray;
-class QDateTime;
-class QString;
-class QStringList;
-class QUrl;
```

Comment:
After some thinking, I removed these forward declarations because those are being used as function return types, I reckon *we should only forward declare the types that only appears as function parameters*.
The reason is, it is unintuitive to have client using this header to include more headers (which include the types that are being returned) in order to use it. For example Qt will include the necessary ones so we (clients) don't have to: https://github.com/qt/qtbase/blob/dev/src/network/ssl/qsslsocket.h#L45

But this is only my opinion, if the majority thinks otherwise then I'll change it back.

Reply:
>The reason is, it is unintuitive to have client using this header to include more headers (which include the types that are being returned) in order to use it.

This is reasonable unless function returns pointer to some type (in that case the type itself can be unused by client code, e.g. we just get the pointer from one function and pass it to the another one).

Reply:
From the other hand some (probable big ) class can be partially used in some places so it don't require many used types there (and we can significantly reduce compile time by having forward declarations of return types instead of including their definitions).

Reply:
For the record: the first post isn't relevant now, still following the usual forward declaration practices for now.